app/include/zmk/hid.h

```diff
@@ -237,5 +255,5 @@ int zmk_hid_consumer_press(zmk_key_t key);
 int zmk_hid_consumer_release(zmk_key_t key);
 void zmk_hid_consumer_clear();
 
-struct zmk_hid_keyboard_report *zmk_hid_get_keyboard_report();
-struct zmk_hid_consumer_report *zmk_hid_get_consumer_report();
+uint8_t *zmk_hid_get_keyboard_report(zmk_hid_report_request_t req_t, uint8_t proto);
```

Comment:
I think I'd be happier with `int zmk_hid_get_keyboard_report(zmk_hid_report_request_t req_t, uint8_t proto, uint8_t **report, size_t *len)`. All the code in usb.c and hog.c care about is there's n bytes of data and data length calculation is better served inside of hid.c.

Reply:
We followed this suggestion when refactoring the patch here: https://github.com/moergo-sc/zmk/commit/8f078c8cc1b296aa8e0cb772e79f4b6edd183b12. Note that that branch isn't yet fully tested; there may still be some regressions.